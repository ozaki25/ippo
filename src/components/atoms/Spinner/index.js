import React from 'react';
import { CircularProgress } from '@material-ui/core';
import styled from 'styled-components';

const Centering = styled.div`
  text-align: center;
`;

const Spinner = props => (
  <Centering>
    <CircularProgress {...props} />
  </Centering>
);

Spinner.displayName = 'Spinner';

export default Spinner;
