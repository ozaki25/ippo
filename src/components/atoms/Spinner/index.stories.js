import React from 'react';
import Spinner from '.';

export default {
  title: 'atoms/Spinner',
};

const props = {};

export const 通常パターン = () => <Spinner {...props} />;
