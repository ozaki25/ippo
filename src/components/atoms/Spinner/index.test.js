import React from 'react';
import { snapshot } from 'src/test/helpers';
import Spinner from '.';

const props = {};

snapshot('Spinner/nomal', <Spinner {...props} />);
