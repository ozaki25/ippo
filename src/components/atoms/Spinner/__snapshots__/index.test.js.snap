// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Spinner/nomal 1`] = `
.c0 {
  text-align: center;
}

<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MockRouter
    createHref={[Function]}
    location=""
    params={Object {}}
    path=""
    push={[Function]}
    replace={[Function]}
    url=""
  >
    <Spinner>
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  text-align: center;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <WithStyles(CircularProgress)>
              <CircularProgress
                classes={
                  Object {
                    "circle": "MuiCircularProgress-circle-7",
                    "circleDisableShrink": "MuiCircularProgress-circleDisableShrink-10",
                    "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-9",
                    "circleStatic": "MuiCircularProgress-circleStatic-8",
                    "colorPrimary": "MuiCircularProgress-colorPrimary-4",
                    "colorSecondary": "MuiCircularProgress-colorSecondary-5",
                    "indeterminate": "MuiCircularProgress-indeterminate-3",
                    "root": "MuiCircularProgress-root-1",
                    "static": "MuiCircularProgress-static-2",
                    "svg": "MuiCircularProgress-svg-6",
                  }
                }
                color="primary"
                disableShrink={false}
                size={40}
                thickness={3.6}
                value={0}
                variant="indeterminate"
              >
                <div
                  className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
                  role="progressbar"
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                >
                  <svg
                    className="MuiCircularProgress-svg-6"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </svg>
                </div>
              </CircularProgress>
            </WithStyles(CircularProgress)>
          </div>
        </StyledComponent>
      </styled.div>
    </Spinner>
  </MockRouter>
</Provider>
`;
